jruby-ext/src/main/java/com/purbon/jrmonitor/monitors/ProcessMonitor.java

```diff
@@ -2,71 +2,127 @@
 
 import com.sun.management.UnixOperatingSystemMXBean;
 
-import javax.management.MBeanServer;
 import java.lang.management.ManagementFactory;
 import java.lang.management.OperatingSystemMXBean;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Objects;
 
 /**
  * Created by andrewvc on 5/12/16.
  */
 public class ProcessMonitor {
     private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
-    private static final MBeanServer platformMxBean = ManagementFactory.getPlatformMBeanServer();
 
-    public class Report {
-        private long memTotalVirtualInBytes = -1;
-        private short cpuSystemPercent = -1;
-        private short cpuProcessPercent = -1;
-        private long cpuMillisTotal = -1;
-        private boolean isUnix;
-        private long openFds = -1;
-        private long maxFds = -1;
-
-        private Map<String, Object> map = new HashMap<>();
+    class Reporter {
+        public Long getOpenFileDescriptorCount() {
+            return Long.valueOf(-1);
+        }
+        public Long getMaxFileDescriptorCount() {
+            return Long.valueOf(-1);
+        }
+        public Long getProcessCpuTime() {
+            return Long.valueOf(-1);
+        }
+        public double getProcessCpuLoad() {
+            return Long.valueOf(-1);
+        }
+        public double getSystemCpuLoad() {
+            return Long.valueOf(-1);
+        }
 
-        public Report() {
-            this.isUnix = osMxBean instanceof UnixOperatingSystemMXBean;
+        public Long getCommittedVirtualMemorySize() {
+            return Long.valueOf(-1);
+        }
+    }
 
-            // Defaults are -1
-            if (this.isUnix) {
-                UnixOperatingSystemMXBean unixOsBean = (UnixOperatingSystemMXBean) osMxBean;;
+    class UnixReporter extends Reporter {
+        private final UnixOperatingSystemMXBean osMxBean;
 
-                this.openFds = unixOsBean.getOpenFileDescriptorCount();
-                this.maxFds =  unixOsBean.getMaxFileDescriptorCount();
+        public UnixReporter(UnixOperatingSystemMXBean osMxBean) {
+            this.osMxBean = osMxBean;
+        }
 
-                this.cpuMillisTotal = unixOsBean.getProcessCpuTime();
-                this.cpuProcessPercent = scaleLoadToPercent(unixOsBean.getProcessCpuLoad());
-                this.cpuSystemPercent = scaleLoadToPercent(unixOsBean.getSystemCpuLoad());
+        public Long getOpenFileDescriptorCount() {
+            try {
+                return this.osMxBean.getOpenFileDescriptorCount();
+            } catch(Throwable e) {
+                return -1L;
+            }
+        }
+        public Long getMaxFileDescriptorCount() {
+            try {
+                return this.osMxBean.getMaxFileDescriptorCount();
+            } catch(Throwable e) {
```

Comment:
The actual exception is a `java.lang.InternalError`, so we could just catch that but if you look at the documentation the OperatingMxBean. There is not a checked exception since gathering metrics is not primordial to running Logstash, I think we should as defensive as possible since we are not planning to allow people to disable it.

I've pinged the user to know if they had the same issue with ES. 

Ref: https://github.com/elastic/elasticsearch/blob/7560101ec73331acb39a042bde6130e59e4bb630/core/src/main/java/org/elasticsearch/monitor/process/ProcessProbe.java

On the other side, I've looked at the OSProble in the elasticsearch source code and they just catch Exception, so ES would probably crash under the same circumstance of https://github.com/elastic/logstash/issues/6296

The more I think, I need to discuss that with someone who might know more about jdk8/freebsd. @jordansissel @untergeek ? Any inputs or guidelines?